### PR TITLE
Add `restrict default` attribute

### PR DESCRIPTION
**Pull request by dgoerger**
The current `templates/default/ntp.conf.erb` hard-codes the `restrict default` attribute:
```
restrict default kod notrap nomodify nopeer noquery
restrict 127.0.0.1 nomodify
restrict -6 default kod notrap nomodify nopeer noquery
restrict -6 ::1 nomodify
```
Per [upstream ntp](http://support.ntp.org/bin/view/Support/AccessRestrictions#Section_6.5.1.1.2.), it's appropriate to set this to `restrict default ignore` in certain use cases of locked-down access control, and it's what our organization has been doing for years per internal security protocol. Would it be possible to create an attribute for this? Upstream indicates there are a few special flags to set to ensure one doesn't bork the config when using `restrict default ignore` (notably exceptions for localhost and the timeserver), but it looks like we're already declaring these in this cookbook, with the exception of timeserver IPv6 exceptions.

This pull request implements that by abstracting this to attribute `node['ntp']['restrict_default']`, and adding the missing IPv6 timeserver exception.

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license           'Apache 2.0'
 description       'Installs and configures ntp as a client or server'
 long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '1.7.0'
+version           '1.8.0'
```
Comment (shortdudey123) [inline metadata.rb]: You should probably let the author bump the version when appropriate to fit their release plans

File: templates/default/ntp.conf.erb [hunk 2]
```diff
@@ -62,14 +62,18 @@ disable monitor
 <% ( node['ntp']['servers'] - node['ntp']['peers'] ).each do |ntpserver| -%>
 <%#   Loop through defined servers, but don't try to upstream ourself %>
 <%   if node['ipaddress'] != ntpserver and node['fqdn'] != ntpserver -%>
+<%# IPv4 %>
 <%  -%>server <%= ntpserver %><% if node['ntp']['server']['use_iburst'] -%> iburst<% end -%><% if node['ntp']['server']['use_burst'] -%> burst<% end -%> minpoll <%= node['ntp']['server']['minpoll'] %> maxpoll <%= node['ntp']['server']['maxpoll'] %>
 <%  -%>restrict <%= ntpserver %> nomodify notrap noquery
+<%# IPv6 %>
+<%  -%>server -6 <%= ntpserver %><% if node['ntp']['server']['use_iburst'] -%> iburst<% end -%><% if node['ntp']['server']['use_burst'] -%> burst<% end -%> minpoll <%= node['ntp']['server']['minpoll'] %> maxpoll <%= node['ntp']['server']['maxpoll'] %>
```
Comment (shortdudey123) [inline templates/default/ntp.conf.erb]: `-6` is only used for DNS resolution. "while a -6 qualifier forces DNS resolution to the IPv6 namespace." ~http://doc.ntp.org/4.2.6/confopt.html
If `node['ntp']['servers']` contains a list of IPv4 addresses, then every server will be defined twice.

**Comment by shortdudey123** [comment]
There should be an entry in `spec/unit/attributes_spec.rb` for the attribute default

**Comment by dgoerger** [comment]
All good points, thank you for the review. Please let me know if there's anything else which should be addressed before it can be merged.

**Comment by gmiranda23** [comment]
Merged.

**Pull request closed without merging**